radix-engine/src/model/resource_manager.rs

```diff
@@ -117,53 +114,52 @@ impl ResourceManager {
         metadata: HashMap<String, String>,
         mut auth: HashMap<ResourceMethod, (AccessRule, Mutability)>,
     ) -> Result<Self, ResourceManagerError> {
+        let mut vault_method_table: HashMap<String, Option<ResourceMethod>> = HashMap::new();
+        vault_method_table.insert("take".to_string(), Some(Withdraw));
+        vault_method_table.insert("put".to_string(), Some(Deposit));
+        if let ResourceType::NonFungible = resource_type {
+            vault_method_table.insert("take_non_fungibles".to_string(), Some(Withdraw));
+        }
```

Comment:
I think we should remove this `if` check. Calling this method on fungible resource should be a runtime error rather than authorisation error.

Reply:
I don't view this as an authorization error but as a method does not exist error. I don't believe such methods should be shown in an abi for example for fungible tokens.

Reply:
Realised that this is the second time we discuss about this. The issue with this design is introduction of polymorphism which we're not sure if it's the right design. We have a single `ResourceManager` blueprint, but different ABIs for `ResourceManager` components. 

If we want distinct method set, we should have `FungibleResourceAddress + FungibleResourceManager` and `NonFungibleResourceAddress + NonFungibleResourceManager`. Although, I didn't see much benefits from this since we're supporting handling non-fungibles in a fungible way.